packages/protocol/contracts/common/Accounts.sol

```diff
@@ -284,40 +352,196 @@ contract Accounts is
    * @dev v, r, s constitute `signer`'s signature on `msg.sender`.
    */
   function authorizeAttestationSigner(address signer, uint8 v, bytes32 r, bytes32 s) public {
-    Account storage account = accounts[msg.sender];
-    authorize(signer, v, r, s);
-    account.signers.attestation = signer;
+    authorizeSignerWithSignature(signer, AttestationSigner, v, r, s);
+    setDefaultSigner(signer, AttestationSigner);
+
     emit AttestationSignerAuthorized(msg.sender, signer);
   }
 
+  /**
+   * @notice Begin the process of authorizing an address to sign on behalf of the account
+   * @param signer The address of the signing key to authorize.
+   * @param role The role to authorize signing for.
+   */
+  function authorizeSigner(address signer, string memory role) public {
+    require(isAccount(msg.sender), "Unknown account");
+    require(
+      isNotAccount(signer) && isNotAuthorizedSignerForAnotherAccount(signer),
+      "Cannot re-authorize address signer"
+    );
+
+    accounts[msg.sender].signerAuthorizations[role][signer] = SignerAuthorization({
+      started: true,
```

Comment:
started is never assigned `false` other than empty storage
do we need it?

Reply:
I believe in completing the authorization in the method below